src/schema/__tests__/city.test.ts

```diff
@@ -0,0 +1,113 @@
+import { runQuery } from "test/utils"
+
+const mockCities = {
+  "sacramende-ca-usa": {
+    slug: "sacramende-ca-usa",
+    name: "Sacramende",
+    coordinates: { lat: 38.5, lng: -121.8 },
+  },
+}
+
+beforeEach(() => {
+  jest.mock("../city/city_data.json", () => mockCities)
+})
+
+describe("City", () => {
+  it("finds a city by its slug", () => {
+    const query = `
+      {
+        city(slug: "sacramende-ca-usa") {
+          name
+        }
+      }
+    `
+
+    return runQuery(query).then(result => {
+      expect(result!.city).toEqual({
+        name: "Sacramende",
+      })
+    })
+  })
+
+  it("returns a helpful error for unknown slugs", () => {
+    expect.assertions(1)
+    const query = `
+      {
+        city(slug: "sacramundo") {
+          name
+        }
+      }
+    `
+    return runQuery(query).catch(e =>
+      expect(e.message).toMatch(/City sacramundo not found in:/)
+    )
+  })
+
+  it("resolves nearby shows", () => {
+    const query = `
+      {
+        city(slug: "sacramende-ca-usa") {
+          name
+          shows {
+            id
+          }
+        }
+      }
+    `
+
+    const mockShows = [{ id: "first-show" }]
+    const mockShowsLoader = jest.fn(() => Promise.resolve(mockShows))
+    const rootValue = {
+      showsLoader: mockShowsLoader,
+      accessToken: null,
+      userID: null,
```

Comment:
I found that I had to add L62-63 to make `tsc` happy, even though I don't think they are necessary for these tests to work.

Reply:
Those properties should be made optional in the `runQuery` interface.